### ISSUE TITLE
musl-translated code needs to be re-audited for correctness 

### ISSUE THREAD

**Issue by expikr** [ziglang/zig#19476]
While reworking `hypot` I spotted [an error](https://github.com/ziglang/zig/commit/4c16f9a3c35b23b9917f2a27b91ba8cd20e6fd82#r140390585) in the auto-translated code deviating from the original behaviour due to function argument mutability, causing the function to produce completely nonsensical answers triggered by the very logic that was specifically meant to guard against those over/underflow cases. 

The behaviour had been overlooked because the tests did not test for the actual over/underflow cases. 

This kind of calls into question for how many lost-in-translation correctness errors arising from the auto-porting process have been overlooked. 

It might be a good idea to keep track of which of the musl code are verified to have been ported correctly, and which of them pends reexamination.

**Comment by tiehuis** [comment]
I agree. The original port was manually-translated and the tests are not comprehensive enough to have full confidence.

A few ideas:
 - Use [libc-test](http://nsz.repo.hu/git/?p=libc-test;a=tree;f=src/math;h=54c835ea8fb4d92d1532a005b996dcf814b69dd4;hb=HEAD) as a basis for extending zig tests. May also be an option to export zig implementation as a c-compatible library to re-use their test harness but less-preferred as we want to test in-tree.
 - Fuzz our implementation against libc. I suspect this would get us very far and uncover most of the obvious issues quite quickly. Extend our test-cases with failures.

**Comment by andrewrk** [comment]
This issue is lacking explicit criteria for which the issue can be closed as completed.
